src/node/internal/crypto.d.ts

```diff
@@ -122,6 +115,22 @@ export class CipherHandle {
   setAuthTag(tag: ArrayBuffer | ArrayBufferView): void;
 }
 
+export class AeadHandle {
+  update(data: ArrayBuffer | ArrayBufferView): ArrayBuffer;
+  final(): ArrayBuffer;
+  setAAD(data: ArrayBuffer | ArrayBufferView, plaintextLength?: number): void;
+  setAutoPadding(autoPadding: boolean): void;
+  getAuthTag(): ArrayBuffer | undefined;
+  setAuthTag(tag: ArrayBuffer | ArrayBufferView): void;
+}
+
+export function newHandle(
+  mode: 'cipher' | 'decipher',
```

Comment:
we can have 2 overloads of this function with 2 different return values and have better types